### PR TITLE
Exclude tests for disabled models in compile stats

### PR DESCRIPTION
**Pull request by NiallRees**
Fix https://github.com/fishtown-analytics/dbt/issues/1804

Checks if the node is a test, and then if it is enabled.

All models enabled:
```
(env) [niallwoodward:~/dev/data-dbt]$ dbt compile                                      
Running with dbt=0.15.0-rc2
Found 328 models, 725 tests, 2 snapshots, 1 analyse, 378 macros, 3 operations, 5 seed files, 104 sources
```

One model disabled before fix (with two tests dependent on the disabled model):
```
(env) [niallwoodward:~/dev/data-dbt]$ dbt compile                                      
Running with dbt=0.14.1
Found 327 models, 735 tests, 2 snapshots, 1 analyse, 374 macros, 3 operations, 5 seed files, 104 sources
```

One model disabled after fix (with two tests dependent on the disabled model):
```
(env) [niallwoodward:~/dev/data-dbt]$ dbt compile                                      
Running with dbt=0.15.0-rc2
Found 327 models, 723 tests, 2 snapshots, 1 analyse, 378 macros, 3 operations, 5 seed files, 104 sources
```

### CODE REVIEW

**Comment by NiallRees** [comment]
CLA is signed @drewbanin

**Comment by drewbanin** [comment]
@cla-bot check

File: core/dbt/compilation.py [hunk 1]
```diff
@@ -54,6 +54,24 @@ def print_compile_stats(stats):
...
+    else:
+        return True
+
+
+def _generate_stats(manifest):
+    stats = defaultdict(int)
+    for node_name, node in itertools.chain(
+            manifest.nodes.items(),
+            manifest.macros.items()):
+        if _node_enabled(node): stats[node.resource_type] += 1
```
Comment (drewbanin) [inline core/dbt/compilation.py]: looks like the `flake8` styleguide checker we use isn't happy about this line:
```
core/dbt/compilation.py:70:31: E701 multiple statements on one line (colon)
```

via https://circleci.com/gh/fishtown-analytics/dbt/19687?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Can you just split this out onto two lines?

Comment (NiallRees) [reply core/dbt/compilation.py]: Sure - thanks

**Comment by NiallRees** [comment]
It looks like the CI failed for an unrelated reason to the pr, are you able to rerun @drewbanin?

**Comment by drewbanin** [comment]
@NiallRees this is great! Thanks very much for your contribution to dbt - this code will ship in dbt v0.16.0 (Barbara Gittings 🎉

**Comment by NiallRees** [comment]
> @NiallRees this is great! Thanks very much for your contribution to dbt - this code will ship in dbt v0.16.0 (Barbara Gittings

Cool thanks - first of many I hope

**Pull request merged**